app/src/main/java/org/mozilla/fenix/components/IntentProcessorType.kt

```diff
@@ -40,7 +40,8 @@ fun IntentProcessors.getType(processor: IntentProcessor?) = when {
         privateCustomTabIntentProcessor == processor -> IntentProcessorType.EXTERNAL_APP
     intentProcessor == processor ||
         privateIntentProcessor == processor ||
-        fennecPageShortcutIntentProcessor == processor -> IntentProcessorType.NEW_TAB
+        fennecPageShortcutIntentProcessor == processor ||
+        webNotificationsIntentProcessor == processor -> IntentProcessorType.NEW_TAB
```

Comment:
this seems like a somewhat complex `when`. should we get a test for it?

Reply:
Tests for checking the returned `IntentProcessorType` existed already. Added a new one for the new processor in `IntentProcessorTypeTest`. Thank you!